langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/DefaultMistralAiHelper.java

```diff
@@ -28,42 +32,62 @@ static List<MistralAiChatMessage> toMistralAiMessages(List<ChatMessage> messages
                 .collect(toList());
     }
 
-    static MistralAiChatMessage toMistralAiMessage(ChatMessage message) {
-        return MistralAiChatMessage.builder()
-                .role(toMistralAiRole(message.type()))
-                .content(toMistralChatMessageContent(message))
-                .build();
-    }
-
-    private static MistralAiRole toMistralAiRole(ChatMessageType chatMessageType) {
-        switch (chatMessageType) {
-            case SYSTEM:
-                return MistralAiRole.SYSTEM;
-            case AI:
-                return MistralAiRole.ASSISTANT;
-            case USER:
-                return MistralAiRole.USER;
-            default:
-                throw new IllegalArgumentException("Unknown chat message type: " + chatMessageType);
-        }
-    }
-
-    private static String toMistralChatMessageContent(ChatMessage message) {
+    private static MistralAiChatMessage toMistralAiMessage(ChatMessage message) {
         if (message instanceof SystemMessage) {
-            return ((SystemMessage) message).text();
+            return MistralAiChatMessage.builder()
+                    .role(MistralAiRole.SYSTEM)
+                    .content(((SystemMessage) message).text())
+                    .build();
         }
 
         if (message instanceof AiMessage) {
-            return ((AiMessage) message).text();
+            AiMessage aiMessage = (AiMessage) message;
+            if (!aiMessage.hasToolExecutionRequests()){
+                return MistralAiChatMessage.builder()
+                        .role(MistralAiRole.ASSISTANT)
+                        .content(aiMessage.text())
+                        .build();
+            }
+
+            List<MistralAiToolCall> toolCalls = aiMessage.toolExecutionRequests().stream()
+                    .map(DefaultMistralAiHelper::toMistralAiToolCall)
+                    .collect(toList());
+
+            return MistralAiChatMessage.builder()
+                    .role(MistralAiRole.ASSISTANT)
+                    .content(aiMessage.text())
```

Comment:
I know that OpenAI return null but MistralAI return "" (empty). We can force that it's null of course but I'm not sure if they plan to keep `content` without value in the future. Basically for that I preferred set same value got from model. It's a simple trade-off.

Below you have an extract of this:
`{"id":"xx","object":"chat.completion","created":12356,"model":"mistral-small-latest","choices":[{"index":0,"message":{"role":"assistant","content":"","tool_calls":[{"function":{"name":"get_weather","arguments":"{\"location\": \"Paris\", \"unit\": \"celsius\"}"}}]},"finish_reason":"tool_calls","logprobs":null}],"usage":{"prompt_tokens":118,"total_tokens":149,"completion_tokens":31}}`

Reply:
Got it. My concern is that some users might do something like this:

```
if (aiMessage.text() != null) {
  // handle text answer
} else {
  // handle tools
}
```

It will work with OpenAI and others, but not with Mistral, so not really portable code.
I think we can add a check: if the `content` is blank (null/empty/whitespaces), use `null` in `AiMessage`. If string is not blank, set it as well, but that is a strange combo.